service/worker/scanner/executions/shard_validator.go

```diff
@@ -0,0 +1,351 @@
+// The MIT License
```

Comment:
i would like to have this logic separated in 2 parts:
* overall control logic
* individual action logic